catalog/views.py

```diff
@@ -48,3 +48,40 @@ def get_context_data(self, **kwargs):
 def author_detail_view(request, primary_key):
     author = get_object_or_404(Author, pk=primary_key)
     return render(request, 'catalog/auhtor_detail.html', context={'author': author})
+
+def index(request):
+    """View function for home page of site"""
+
+    # Generate counts of the main objects
+    num_books = Book.objects.count()
+    num_instances = BookInstance.objects.count()
+    num_instances_available = BookInstance.objects.filter(status__exact='a').count()# Available books (status = 'a')
+    num_authors = Author.objects.count()
+    
+    # Number of visits to this view, as counted in the session variable.
+    num_visits = request.session.get('num_visits', 1)
```

Comment:
- Cơ chế hoạt động của session trong django?
- Method `get` hoạt động ntn?

Reply:
method `get` hoạt động với cú pháp `get(key, default)`
- `key` là session, một khóa mà mình muốn tìm
- `default` là giá trị mặc định

Thì trong đoạn code 
```
num_visits = request.session.get('num_visits', 1)
 request.session['num_visits'] = num_visits + 1
```

- Nếu `num_visits` tồn tại trong `session`, giá trị của nó sẽ được trả về. Nếu không tồn tại, giá trị mặc định là 1 sẽ được trả về.
- Ngay khi chưa truy cập, thì `num_visits` sẽ được mặc định là 1, và nó sẽ được in ra màn hình khi người dùng truy cập. Và khi người dùng truy cập lần 1, giá trị `num_visits` sẽ được +1 để in ra màn hình cho lần tiếp theo người dùng truy cập. 

=>Có nghĩa là 'num_visits' trong session sẽ lớn hơn số lần truy cập thực tế 1 lần để lấy cái đó gán cho biến num_visits ở lần truy cập tiếp theo.

Reply:
Tính năng này là đếm số lần ĐÃ tuy cập, KO đếm lần truy cập hiện tại
=> Giá trị `0` ở đây mới chuẩn e nhé

Reply:
ra là vậy, em cứ tưởng là đếm số lần truy cập hiện tại
Em đã sửa ạ